### ISSUE TITLE
[Question] Set bitrate

### ISSUE THREAD

**Issue by alexsafe** [pedroSG94/RootEncoder#1690]
Hi Pedro,

I'm flagging this as just a question for now, because I think it relates to the general way in which the bitrate is set. When setting the bitrate either by prepareVideo or setVideoBitrateOnFly in the stream I see that the bitrate kind of matches the one I set but when I move the camera, so where there is "action" the bitrate increases well above the set value. If the camera faces a static scene only  then the bitrate I set is reached.
Is this the expected behaviour?

**Comment by pedroSG94** [comment]
Hello,

Sorry for late response. I'm on holidays

This is expected but depend a bit of the encoder used. The H264 codec reduce the bitrate if the image current image is similar to the previous image because repeat values to avoid increase the size but when te image is different the bitrate increase because you need send new info to properly create the image.

You can check if this way:
If you put your hand in the camera to full cover it and create a black image the bitrate reduce to a lower values because all images are similar to the previous one but if you move the devices fast all time the bitrate increase a higher values.

Also, the difference is the bitrate could change if you use a codec that support CBR but not all devices support it.

**Comment by alexsafe** [comment]
hehe yeah no problem same here. 
The behavior you described is the same for me too. I was hoping to find a solution to better control the bitrate. like streaming 720p at 800kbps or 1500 kbps for instance. But if I set it for 800 I would like to not have it spike at 2000 kbps or something :). 

For now my observations seem to indicate that the stream will try to send at the set bitrate (say 800) as long as there isn't much happening in front of the camera (like you also described). But then if you move something in front of the camera the bitrate will go as high as the network allows it for the given resolution. The use-case is that the network speed is very variable so I'd like to have more controls over those spikes. Is there something I could use for this?

**Comment by pedroSG94** [comment]
Hello,

Unfortunately you can't do anything about it. You can try use other codec like H265 and check if the result is better but it should be similar. Anyway, H265 can get the same quality with about 20-30% less bitrate so the spikes should be less to

**Comment by alexsafe** [comment]
Yeah I thought so. With H265 it's the problem on the server side, the viewers need to be on Safari

**Comment by pedroSG94** [comment]
Maybe you can transcode from H265 to H264 in server side to solve the problem. Fews servers like media MTX can you a ffmpeg command to do it.

**Comment by alexsafe** [comment]
yeah that's something we are low-key exploring thanks. (Sorry I somehow missed your reply)

Perhaps I should play with the adaptBitrate() method too. The end goal was force the stream to have a decent enough bitrate with a higher resolution.
